blog/invictus-logic-apps-testing-framework.md

```diff
@@ -0,0 +1,122 @@
+
+Up until now, when building interfaces on top of Azure Logic Apps, you were forced to manually (re)test every implemented feature because there simply wasn't anything available to help you on your way.  
+
+While this approach works fine, we've found that it has a few flaws:  
+- New people on your team don't have an easy way to understand how flows should work  
+- It's cumbersome since you cannot easily automate it to guarantee quality  
+- It takes time & effort which ends up in a higher engineering & testing cost, money you'd like to spend elsewhere  
+
+Since developing the Invictus Methodology, which includes patterns, best practices and several templates, Codit has been investing in building a solid foundation enabling us to quickly kick-off new projects, while ensuring a high level of quality. The biggest thing which, until now, was still lacking, was the ability to include test-cases into those projects.  
+Because of this, we started working on a framework that allowed us to easily enable/disable, trigger and even modify existing Azure Logic Apps. But, since the operations are not enough to write full-blown test-scenarios, we extended this list to ensure you would also be able to monitor the executed Azure Logic App runs.  
+
+All these operations have now been forged into the Invictus Test Framework for Azure Logic Apps which we used internally for our customers and have proven to be effective.  
+Today, **we are happy to announce that we are open-sourcing Invictus Test Framework for Azure Logic Apps on GitHub** and it is now available on NuGet for you to use on your projects!  
+
+Get started very easily :  
+```shell
+> Install-Package Invictus.Testing.LogicApps
+```
+
+### Authentication
+When you want to use this package to track of control your Logic Apps, you will have to authenticate against the subscription containing said Logic Apps of course.  
+This can be done by re-using the existing service principal, which has been created to connect your Azure DevOps project to the Azure subscription. However, seeing as this service principal typically has contributor-access to the entire subscription, which is quite an overkill for testing-purposes, it would be advised to create a new service principal, entirely dedicated to the testing scenarios.  
+This test-service principal will need to get the "*Logic App Contributor*"-role assigned, as any other role would be insufficient to update your Logic Apps (to enable static results, for instance).  
+
+But what exactly do you need to specify in order to really authenticate?  
+Well, below is an overview of the required information:
+
+```csharp
+// The Tenant ID as found in the AAD properties
+string tenantId = "my-tenant-id";
+// The ID of the subscription containing your Logic Apps.
+string subscriptionId = "my-subscription-id";
+// The Client ID of the service principal to be used to authenticate.
```

Comment:
```suggestion
// The application id of the service principal to be used to authenticate.
```

Reply:
Or am I the only one never knowing if it's the app or object id? :p

Reply:
Argh, constantly mixing them up.